modules/rgbd/src/opencl/tsdf.cl

```diff
@@ -214,7 +214,10 @@ inline float3 getNormalVoxel(float3 p, __global const float2* volumePtr,
     }
 
     //gradientDeltaFactor is fixed at 1.0 of voxel size
-    return fast_normalize(vload3(0, an));
+    float3 n = vload3(0, an);
+    float Norm = sqrt(n.x*n.x + n.y*n.y + n.z*n.z);
+    return Norm < 0.0001 ? (float3)(0,0,0) : n / Norm;
```

Comment:
`nan((uint)0)` should be returned when norm is too small, see line 185 for example

Reply:
done